CONTRIBUTING.md

```diff
@@ -0,0 +1,29 @@
+The [Battle for Bikini Bottom Decompilation Project](https://github.com/bfbbdecomp/bfbb) is happy to accept all contributions of all size and scope! 
+
+This documentation outlines our project contribution guidelines. Following these guidelines allows the maintainers review changes faster and give better feedback, and helps to ensure that future modding and PC port efforts are made as simple as possible.
+
+In addition to decompilation efforts, this project also welcomes tooling and [tutorial/knowledge resource](https://github.com/bfbbdecomp/bfbb-decomp-guide) contributions! Check out [our organization page](https://github.com/bfbbdecomp) to see all of the projects we welcome contributions to.
+
+## Contribution Guidelines
+
+Please respect the following guidelines:
+- Ensure your code is formatted using the project's clang-format autoconfiguration ([VS Code Plugin](https://marketplace.visualstudio.com/items?itemName=xaver.clang-format)).
+- Keep pull requests scoped to a single compilation unit when possible. Defining functions in other files is encouraged, however, in cases where the function is used by the compilation unit you're contributing to.
+- A good-faith attempt to clean up code generated by decompiler tools (m2c, Ghidra) is expected, using DWARF data and project code conventions as appropriate.
+- Encourage and support other contributors, especially new ones! Decompiling games is hard and difficult work, but sharing knowledge and supporting fellow decompers makes that work a lot easier and more enjoyable for all of us.
+
+## Code Conventions
+- Avoid using pointer math for unknown struct members. Instead, populate the struct with padding and the appropriate type using `unk_<offset>` as the naming convention. For example, define an unknown float member at offset 0xF0 as `F32 unk_F0`.
+- Use the `TRUE`/`FALSE` macros in contexts where conditions use the value of a non-bool data type (S32, U8) as a boolean-like.
+- Use the `NULL` macro when assigning `0`/`0x0` to a pointer value.
+- Do not explicitly define enum values, unless the starting value is non-zero or the increment is non-standard.
+- Use PowerPC naming conventions to provide variable names for values which have no symbols contained in the available DWARF data, i.e:
```

Comment:
I think I'd prefer a bit more nuance here. I personally try to reason about what the symbol actually means/does before falling back to a placeholder name

also in sufficiently complex functions, registers may be reused if the variable's lifetimes don't overlap.

Otherwise I think I'm fine with this convention, so I'd just phrase it as a first choice for a placeholder name or something like that.

Reply:
I support that premise. Maybe something like..?
```suggestion
- Use PowerPC naming conventions when creating placeholder variable names. Placeholders should only be used when no DWARF symbol data is available and the variable does not have a name that can be reasonably assigned based on its usage.
```

Additionally, I feel that especially with complex functions, even if there is overlap and the variable is used multiple times, it makes the code a little easier to decompile since it reads closer to its assembly equivalent. I'm open to discussion on that point, though.

Reply:
That's a good point, the type might vary from usage to usage. Maybe something like r3_2 or r3_3 for subsequence declarations? Although it starts to look a little bit ugly - but like you mentioned this case isn't very common so we're really discussing naming theory for general PowerPC decomp projects. I'm not aware of a standard.

Reply:
That's fine with me.